src/lib/providers/remoteFsProvider.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  TreeDataProvider,
+  TreeItem,
+  Uri,
+  TreeItemCollapsibleState,
+  ThemeIcon,
+  EventEmitter,
+  Event,
+} from 'vscode';
+import { FileLink, RemoteFsDirectory } from '../types';
+
+const {
+  getDirectoryContentsByPath,
+} = require('@hubspot/cli-lib/api/fileMapper');
+const { getPortalId } = require('@hubspot/cli-lib');
+const { FOLDER_DOT_EXTENSIONS } = require('@hubspot/cli-lib/lib/constants');
+
+function isPathFolder(path: string) {
+  const splitPath = path.split('/');
+  const fileOrFolderName = splitPath[splitPath.length - 1];
+  const splitName = fileOrFolderName.split('.');
+
+  return !(
+    splitName.length > 1 && FOLDER_DOT_EXTENSIONS.indexOf(splitName[1]) === -1
+  );
+}
+export class RemoteFsProvider implements TreeDataProvider<FileLink> {
+  private _onDidChangeTreeData: EventEmitter<
+    FileLink | undefined | null | void
+  > = new EventEmitter<FileLink | undefined | null | void>();
+  readonly onDidChangeTreeData: Event<void | FileLink | null | undefined> =
+    this._onDidChangeTreeData.event;
+
+  refresh(): void {
+    this._onDidChangeTreeData.fire();
+  }
```

Comment:
This + above two declarations set up refreshing, when the refresh command gets fired this refills the view with getChildren()